src/crewai/agent.py

```diff
@@ -120,9 +121,14 @@ class Agent(BaseAgent):
         default="safe",
         description="Mode for code execution: 'safe' (using Docker) or 'unsafe' (direct execution).",
     )
+    knowledge: Optional[Dict[str, Any]] = Field(
```

Comment:
Shouldn't the type for knowledge be `Optional[List[BaseKnowledgeSource]`?

Reply:
Type "Knowledge" is incompatible with type "Dict[str, Any] | None"
    "Knowledge" is incompatible with "Dict[str, Any]"
    "Knowledge" is incompatible with 
    
    Getting this error from line 251.

Reply:
No, because like memory you can set your embedder config from there as well. Or maybe there’s a better way of handling this